backend/PythonClient/multirotor/mission/abstract/abstract_mission.py

```diff
@@ -25,6 +26,26 @@ def __init__(self, target_drone="Default"):
         self.objects = [self.client.simGetObjectPose(i) for i in self.all_drone_names]
         self.states = [self.client.getMultirotorState(i) for i in self.all_drone_names]
         self.client.enableApiControl(True, vehicle_name=target_drone)
+        
+        #Initialize a google cloud client and bucket
+        self.gcs_client = storage.Client()
+        self.gcs_bucket_name = gcs_bucket_name
+        
+        self.create_bucket_if_not_exists(location)
+        
+    def create_bucket_if_not_exists(self, location="US"):
+        try:
+            # Check if the bucket already exists
+            bucket = self.gcs_client.lookup_bucket(self.gcs_bucket_name)
+            if bucket is None:
+                print(f"Bucket '{self.gcs_bucket_name}' does not exist. Creating it now...")
+                # Create the bucket
+                bucket = self.gcs_client.create_bucket(self.gcs_bucket_name, location=location)
+                print(f"Bucket '{self.gcs_bucket_name}' created in location '{location}'.")
+            else:
+                print(f"Bucket '{self.gcs_bucket_name}' already exists.")
+        except Exception as e:
+            print(f"Error creating bucket: {str(e)}")
 
```

Comment:
GCS client and bucket creation in abstract_mission.py should not be handled by each mission; this is done globally during initialization.